pkg/apis/core/validation/validation.go

```diff
@@ -3873,9 +3873,37 @@ func ValidateContainerUpdates(newContainers, oldContainers []core.Container, fld
 			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("image"), ctr.Image, "must not have leading or trailing whitespace"))
 		}
 	}
+
+	// validate updated container probe
+	for i, ctr := range newContainers {
+		allErrs = append(allErrs, validateProbeUpdates(ctr.StartupProbe, oldContainers[i].StartupProbe, fldPath.Index(i).Child("startupProbe"))...)
+		allErrs = append(allErrs, validateProbeUpdates(ctr.LivenessProbe, oldContainers[i].LivenessProbe, fldPath.Index(i).Child("livenessProbe"))...)
+	}
+
 	return allErrs, false
 }
 
+func validateProbeUpdates(newProbe, oldProbe *core.Probe, fldPath *field.Path) (allErrs field.ErrorList) {
```

Comment:
this is a bigger question than just the negative value validation, but won't a client with an old v1.Pod definition drop probe TerminationGracePeriodSeconds field values when it does a read/update/write and fail the immutability check?

should we be copying TerminationGracePeriodSeconds values from the old pod's probes in if the new pod has nil values for those to keep existing clients working?